plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/binary/KSPropertySetterDescriptorImpl.kt

```diff
@@ -8,22 +8,40 @@ package org.jetbrains.kotlin.ksp.symbol.impl.binary
 import org.jetbrains.kotlin.descriptors.PropertySetterDescriptor
 import org.jetbrains.kotlin.ksp.symbol.*
 import org.jetbrains.kotlin.ksp.symbol.impl.KSObjectCache
+import org.jetbrains.kotlin.ksp.symbol.impl.findPsi
+import org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSPropertyDeclarationImpl
+import org.jetbrains.kotlin.ksp.symbol.impl.kotlin.KSPropertyDeclarationParameterImpl
 import org.jetbrains.kotlin.ksp.symbol.impl.toFunctionKSModifiers
 import org.jetbrains.kotlin.ksp.symbol.impl.toKSModifiers
+import org.jetbrains.kotlin.psi.KtParameter
+import org.jetbrains.kotlin.psi.KtProperty
 
 class KSPropertySetterDescriptorImpl private constructor(val descriptor: PropertySetterDescriptor) : KSPropertySetter {
     companion object : KSObjectCache<PropertySetterDescriptor, KSPropertySetterDescriptorImpl>() {
         fun getCached(descriptor: PropertySetterDescriptor) = cache.getOrPut(descriptor) { KSPropertySetterDescriptorImpl(descriptor) }
     }
 
-    override val origin = Origin.CLASS
+    override val origin by lazy {
```

Comment:
Can this and owner be hoisted to `abstract KSPropertyAccessorDescriptorImpl`, and shared with getter?